tests/test_subdomains.py

```diff
@@ -628,3 +628,101 @@ class Dummy(SubDomainSet):
         assert x.is_Parallel
         assert y.is_Parallel
         assert z.is_Parallel
+
+
+class TestSubDomain_w_condition(object):
+
+    def test_condition_w_subdomain(self):
+
+        shape = (10, )
+        grid = Grid(shape=shape)
+        x, = grid.dimensions
+
+        class Middle(SubDomain):
+            name = 'middle'
+
+            def define(self, dimensions):
+                return {x: ('middle', 2, 4)}
+
+        mid = Middle()
+        my_grid = Grid(shape=shape, subdomains=(mid, ))
+
+        f = Function(name='f', grid=my_grid)
+
+        sdf = Function(name='sdf', grid=my_grid)
+        sdf.data[5:] = 1
+
+        condition = Lt(sdf[mid.dimensions[0]], 1)
+
+        ci = ConditionalDimension(name='ci', condition=condition,
+                                  parent=mid.dimensions[0])
```

Comment:
mid.dimensions[0] is a root dimension already, do we actually need this test? 

IOW, would this test actually fail in current master?

Reply:
https://github.com/devitocodes/devito/pull/2050#issuecomment-1408497079